Network Class/Network.h

```diff
@@ -18,55 +20,161 @@ typedef std::vector<int> Vector;
 
 class Network
 {
-	public:
-							Network				(istream& = cin);
-							Network				(const string&, const string&);
-							Network				(const string&);
-							~Network			();
-		void				train				();
-		void				classify			(const string&);
-
-	private:
-		double				learningRate;
-		int					batchSize;
-		int					epochs;
-		int 				numLayers;
-		
-		VMatrix				weights;
-		VMatrix				sumNablaW;
-		VMatrix				biases;
-		VMatrix				activations;
-		VMatrix				weightedInputs;
-		VMatrix				errors;
-		VMatrix				sumNablaB;
-		
-		Vector				miniBatchIndices;
-		Vector				layerSizes;
-	
-	//we do need the string names to be the class data members. Just realized, we need to write the names of the files in the Previous_Network_File.
-		string				trainingDataFilename;
-		string				expectedValuesFilename;
-	
-		ifstream			trainingDataInfile;
-		ifstream			expectedValuesInfile;
+public:
+
+	//**********Constructors and Destructors*******************
+						Network();
+						Network(const string&, const string&);	
+						Network(const string&);					
+						~Network();
+
+	//**********Public Accessible Methods**********************
+	void				forwardProp(int, ifstream &);
+	int					SGD();
```

Comment:
Will the user ever call `SGD()`? They'd need access to `miniBatchIndices` then.

Reply:
Well, I was doubtful about that! We can make it a private function and have a accessor function for numCorrect, which will let user know how many correct outputs were found during each epoch!

Reply:
Or, rather it can be public. The user is the one who enters the batch size, so the user can shuffle it in the way the user wants (or we can have the FYShuffle() be public) and pass it as a parameter to `SGD (int []) `

Reply:
(we can add that parameter to SGD())

Reply:
Wasn't there an idea floating around to let `train()` return a `struct` of relevant data? At the moment `numCorrect` isn't a class attribute. An accessor function won't work unless that's the case. Eventually, we really need to figure out how to allow the user to write certain functions. Ideally, `FYshuffle()` won't be inside of the class for reasons of flexibility you just mentioned. Is the end goal of the proposed update is to allow the user to write their own `train()`? Anyway, it looks easy enough to do. However, the new parameter should be `SGD(std::vector<int>)` since that's what we're using in the code!

@yonatanmg 
@patelami3431 
*On a **really** important side-note*:
I don't know about you, but for me the class didn't work *at all* when copy-pasted into Visual Studio. I've had to tinker with it over the past couple of days to get it working. Even then, we've had to use shortcuts to allow for the rest of the system to work. What I've got now doesn't look much like what's in the repo. **The upshot of this is that the code on GitHub doesn't work as we discussed in the meeting with Potter, Slater, and Gutzler.**

I'd really advise you get the GitHub extension, create a branch for your updates (**NOT** master) that we can look at and pull from as we develop the master branch. **I get the impression that we're all looking at different Network.h and Network.cpp files**, and so our decisions are guided separately, outside of the view of one another.